tests/components/intellifire/conftest.py

```diff
@@ -17,35 +41,241 @@ def mock_setup_entry() -> Generator[AsyncMock]:
+    async def side_effect_func(*args, **kwargs):
+        if not hasattr(side_effect_func, "called"):
+            side_effect_func.called = True
+            raise LoginError
+        # Simulate async function
+
+    with patch(
+        "homeassistant.components.intellifire.config_flow.IntelliFireCloudInterface.login_with_credentials",
+        new_callable=AsyncMock,
+    ) as mock_login:
+        mock_login.side_effect = side_effect_func
+        yield mock_login
+
+
+@pytest.fixture
+def mock_poll_local_fireplace() -> Generator[AsyncMock, None, None]:
+    """Mock polling of local fireplace."""
+    with patch(
+        "homeassistant.components.intellifire.config_flow.IntelliFireAPILocal.poll",
+        new_callable=AsyncMock,
+    ) as mock_poll:
+        # mock_login.side_effect = login_side_effect
+        yield mock_poll
+
+
+@pytest.fixture
+def mock_cloud_poll() -> Generator[AsyncMock, None, None]:
+    """Mock a successful cloud poll call."""
+    with patch(
+        "homeassistant.components.intellifire.__init__.UnifiedFireplace.perform_cloud_poll",
+        new_callable=AsyncMock,
+    ) as mock_poll:
+        # mock_login.side_effect = login_side_effect
+        yield mock_poll
+
+
+@pytest.fixture
+def mock_connectivity_test_pass_pass() -> Generator[AsyncMock, None, None]:
+    """Mock both Cloud and Local Connectivity."""
+    with patch(
+        "homeassistant.components.intellifire.__init__.UnifiedFireplace.async_validate_connectivity",
+        new_callable=AsyncMock,
+    ) as mock_connectivity:
+        mock_connectivity.return_value = (True, True)
+        yield mock_connectivity
+
+
+@pytest.fixture
+def mock_connectivity_test_fail_fail() -> Generator[AsyncMock, None, None]:
+    """Mock both Cloud and Local Connectivity."""
+    with patch(
+        "homeassistant.components.intellifire.__init__.UnifiedFireplace.async_validate_connectivity",
+        new_callable=AsyncMock,
+    ) as mock_connectivity:
+        mock_connectivity.return_value = (False, False)
+        yield mock_connectivity
+
+
+@pytest.fixture
+def mock_connectivity_test_fail_fail_then_pass_pass() -> (
+    Generator[AsyncMock, None, None]
+):
+    """Mock both Cloud and Local Connectivity."""
+    return_values = [(False, False), (True, True), (True, True), (True, True)]
+
+    with patch(
+        "homeassistant.components.intellifire.__init__.UnifiedFireplace.async_validate_connectivity",
+        new_callable=AsyncMock,
+    ) as mock_connectivity:
+        mock_connectivity.side_effect = return_values
+        yield mock_connectivity
+
+
+@pytest.fixture
+def mock_poll_local_fireplace_exception() -> Generator[AsyncMock, None, None]:
+    """Mock polling a fireplace that isn't a fireplace."""
+    with patch(
+        "homeassistant.components.intellifire.config_flow.IntelliFireAPILocal.poll",
+        new_callable=AsyncMock,
+    ) as mock_poll:
+        mock_poll.side_effect = ConnectionError
+        yield mock_poll
+
+
```

Comment:
Why do we have so many mocks? Can we mock the entrypoint where we use the library? Checkout `airgradient` for an example

Reply:
Because when this was written I think I was dumb at writing mocks

Reply:
I think I've significantly improved the number of mocks now. Other than config entries (which you suggested maybe moving into the tests... and I'll look at that) they should be significantly reduced